ballerina/modules/wssec/tests/ws_security_tests.bal

```diff
@@ -0,0 +1,1017 @@
+        servicePublicKey: serverPublicKey,
+        x509Token: X509_PUBLIC_CERT_PATH_2
+    };
+
+    xml securedEnvelope = check applySymmetricBinding(envelope, symmetricBinding);
+    string envelopeString = securedEnvelope.toString();
+    byte[] signedData = check getSignatureData(securedEnvelope);
+
+    boolean validity = check crypto:verifyRsaSha256Signature((envelope/<soap:Body>/*).toString().toBytes(), signedData, publicKey);
+    test:assertTrue(validity);
+
+    byte[] encData = check getEncryptedData(securedEnvelope);
+    byte[] decryptDataResult = check decryptData(encData, RSA_ECB, publicKey);
+    test:assertEquals((envelope/<soap:Body>/*).toString(), check string:fromBytes(decryptDataResult));
+
+    assertEncryptedSymmetricKey(envelopeString);
+    assertUsernameToken(envelopeString, DIGEST);
+    assertTimestampToken(envelopeString);
+    assertSignatureWithoutX509(envelopeString);
+    assertEncryptedPart(envelopeString);
+}
+
+@test:Config {
+    groups: ["username_token", "signature", "symmetric_binding", "outbound_config"]
+}
+function testSymmetricBindingWithOutboundConfig() returns error? {
+    xml envelope =
+    xml `<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+            <soap:Body><person></person></soap:Body>
+        </soap:Envelope>`;
+    xmlns "http://schemas.xmlsoap.org/soap/envelope/" as soap;
+
+    crypto:KeyStore serverKeyStore = {
+        path: X509_KEY_STORE_PATH,
+        password: KEY_PASSWORD
+    };
+    crypto:PublicKey serverPublicKey = check crypto:decodeRsaPublicKeyFromTrustStore(serverKeyStore, KEY_ALIAS);
+
+    crypto:KeyStore keyStore = {
+        path: KEY_STORE_PATH,
+        password: KEY_PASSWORD
+    };
+    crypto:PrivateKey symmetricKey = check crypto:decodeRsaPrivateKeyFromKeyStore(keyStore, KEY_ALIAS, KEY_PASSWORD);
+    crypto:PublicKey publicKey = check crypto:decodeRsaPublicKeyFromTrustStore(keyStore, KEY_ALIAS);
+
+    SymmetricBindingConfig symmetricBinding = {
+        signatureAlgorithm: RSA_SHA256,
+        encryptionAlgorithm: RSA_ECB,
+        symmetricKey: symmetricKey,
+        servicePublicKey: serverPublicKey
+    };
+
+    xml securedEnvelope = check applySymmetricBinding(envelope, symmetricBinding);
+    string envelopeString = securedEnvelope.toString();
+
+    OutboundSecurityConfig outboundConfig = {
+        verificationKey: publicKey,
+        signatureAlgorithm: RSA_SHA256,
+        decryptionAlgorithm: RSA_ECB,
+        decryptionKey: publicKey
+    };
+
+    crypto:PrivateKey|crypto:PublicKey? privateKey = outboundConfig.decryptionKey;
+    if privateKey !is () {
+        byte[] encData = check getEncryptedData(securedEnvelope);
+        byte[] decryptDataResult = check decryptData(encData, RSA_ECB, privateKey);
+        string decryptedBody = "<soap:Body >" + check string:fromBytes(decryptDataResult) + "</soap:Body>";
+        envelopeString = regex:replace(envelopeString, string`<soap:Body .*>.*</soap:Body>`, decryptedBody);
```

Comment:
regex module is now deprecated. shall we use `regexp`

Reply:
Fixed in [1de6750](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/1de6750f3a32d73e90b0ade50476774f2541fccf)